### PR TITLE
feat(date-helpers): added support for formatting date in short ordinal form

### PR DESCRIPTION
**Pull request by EngM7mdIbrahim**
### Screenshots

### CODE REVIEW

File: src/date.ts [hunk 1]
```diff
@@ -39,3 +39,39 @@ export const formatDateTime = (
...
 ): string => new Date(date).toLocaleDateString(localTimezone, options)
+
+/**
+ *
+ * @function
+ * @param {string | Date} date - The date string that needs to be returned as a Date object.
+ * @returns {Date} - A date that represents the new date object.
+ */
+
+export function getDate(rawDate: string | Date): Date {
```
[the remark this answers is not in the document]
Comment (guy-shahine) [reply src/date.ts:50]: Let's rename the method to `parseDate`? and let's add date validation on both the string and passed date. Let's write unit tests for these two methods

**Review by guy-shahine — CHANGES_REQUESTED** [review]

File: src/date.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (EngM7mdIbrahim) [reply src/date.ts]: addressed

File: __tests__/date.test.ts [hunk 2]
```diff
@@ -51,4 +51,77 @@ describe("date", () => {
 
     expect(actual).toBe(expected)
   })
+
+  it("parseDate with not a date string", () => {
```
Comment (tsatsujnr139) [inline __tests__/date.test.ts:55]:
```suggestion
  it("parseDate with invalid date string", () => {
```

File: __tests__/date.test.ts [hunk 3]
```diff
@@ -51,4 +51,77 @@ describe("date", () => {
...
+    const dateParser = () => parseDate("not date string")
+    const expected = 'Cannot parse passed date: "not date string"'
+    expect(dateParser).toThrow(expected)
+  })
+
+  it("parseDate with a date string", () => {
```
Comment (tsatsujnr139) [inline __tests__/date.test.ts:61]:
```suggestion
  it("parseDate with a valid date string", () => {
```

Comment (tsatsujnr139) [reply __tests__/date.test.ts:61]: Apply similar naming to the others named similarly

**Review by tsatsujnr139 — CHANGES_REQUESTED** [review]

File: __tests__/date.test.ts [hunk 4]
```diff
@@ -51,4 +51,77 @@ describe("date", () => {
 
     expect(actual).toBe(expected)
   })
+
+  it("parseDate with invalid date string", () => {
+    const dateParser = () => parseDate("not date string")
```
Comment (guy-shahine) [inline __tests__/date.test.ts:56]:
```suggestion
    const dateParser = () => parseDate("not a date string")
```

File: __tests__/date.test.ts [hunk 5]
```diff
@@ -51,4 +51,77 @@ describe("date", () => {
...
+    const expected = 'Cannot parse passed date: "not date string"'
```
Comment (guy-shahine) [inline __tests__/date.test.ts:57]:
```suggestion
    const expected = 'Cannot parse passed date: "not a date string"'
```

File: __tests__/date.test.ts (hunk 3 above)
Comment (EngM7mdIbrahim) [reply __tests__/date.test.ts]: addressed

File: __tests__/date.test.ts (hunk 4 above)
Comment (EngM7mdIbrahim) [reply __tests__/date.test.ts]: addressed

File: __tests__/date.test.ts (hunk 5 above)
Comment (EngM7mdIbrahim) [reply __tests__/date.test.ts]: addressed

**Review by guy-shahine — APPROVED** [review]

**Review by tsatsujnr139 — APPROVED** [review]

**Pull request merged**